www/src/pages/semesters/api/semesters.ts

```diff
@@ -0,0 +1,31 @@
+import type { APIRoute } from 'astro';
+import { z } from 'zod';
+import { Semester } from '@backend/semester';
+
+const schema = z.object({
+    year: z.number(),
+    type: z.enum(['summer', 'winter']),
+    startDate: z.coerce.date(),
```

Comment:
When working with dates we need to be extra careful about time zone issues. When constructing a `Date` from a string we need to keep in mind how it is parsed (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#date_time_string_format). `node-pg` reads the date from `Date` class in local time (using `getDate`, `getMonth` and `getFullYear` functions instead of their UTC counterparts). However when parsing a date-only string in `YYYY-MM-DD` format the string is interpreted as UTC time (due to a historical spec error). A date-time string in `YYYY-MM-DDTHH:mm:ss.sss` format will be interpreted as local time (which is what we want). Therefore we need to apply a more complex transformation:
```suggestion
    startDate: z
        .string()
        .date()
        .transform(str => {
            // When the time zone offset is absent, date-only forms are interpreted as a UTC time
            // and date-time forms are interpreted as local time
            return new Date(`${str}T00:00:00`);
        }),
```